### DISCUSSION TITLE
Polling with heavy data tables (optimization ideas needed)

### DISCUSSION THREAD

**Discussion by petsoukos** [livewire/livewire#5295] [1. Help]
Hello,

I recently switched to Livewire as a go to first when I extend a company internal admin system with new views.  We're dealing with loads of data and relationships between various models. This data needs to be displayed to operators for monitoring and doing company stuff with it.

This is totally unrelated to this issue, but it is related in a way that this data constantly changes and "flows" in the database. There is a background process generating the raw data which gets transformed and ingested. That's why we requiring the polling feature of Livewire so as to keep the operator informed of changes in the data. Just imagine it as looking at a stock trading table.

Optimizations already done for calculating the data required to be displayed in the data table, and the 8~10s loadtime is not bad for our needs, but how can we decouple the calculation of the data from the view as to make the view more responsive and feel "real time-ish". So far I haven't found a good way or an elegant way to do that.

Currently in the prototype phase, the data comes from one of the models paginated, lets call it `Flights`, and there is a minimum of 6 columns in the table each with different statistics. This makes this Livewire component look unresponsive, because it has lots of buttons and checkboxes and inputs that need to be clicked, selected, typed into while binding them to properties. There's also a log being polled every 2s (default) and displayed in a panel on the same page (I will take it out to a separate component). 

The problem with paginated views and Livewire is that it requires the paginator/data to be passed into the view in the render() method. Doing that, the rest of the methods in the component are unaware of which `Flights` are loaded in the view that's why we currently have to calculate the 6 data columns in the render method and polling makes it bleh.. unresponsive. This data cant be cached as it needs to be real time, but if we could only update certain columns only, even with events or push messages.

Maybe I didn't get the pagination documentation, but I cant pass a paginator with any other way without converting it into an array (it loads the first time just fine, but then the polling kicks in and the $dataVariable becomes an array and then there is no ->links() method).

Sorry for the long post, but how would you approach this issue? Making a Paginated Datatable that requires dynamically updating rows of data? Any update changes the statistics on all columns.

Any way any ideas welcome. Sorry for the long post.

**Comment by joshhanley** [comment]
@petsoukos I would probably make use of a computed property or even just a standard method that returns the paginated results, which can be called in the render method.

Inside the computed property/ method I would make use of caching to cache the results, which should speed up response times.

Then finally I would have another method, which busts the cache, which you can get the poll to call on whatever interval you need, like `wire:poll="bustCache()"`.

Hope this helps!
